elephant_vending_machine/views.py

```diff
@@ -8,48 +8,85 @@
 # Circular import OK here. See https://flask.palletsprojects.com/en/1.1.x/patterns/packages/
 # pylint: disable=cyclic-import
 from datetime import datetime
+import importlib.util
 import os
 import subprocess
 from subprocess import CalledProcessError
 from flask import request, make_response, jsonify
 from werkzeug.utils import secure_filename
 from elephant_vending_machine import APP
 from .libraries.experiment_logger import create_experiment_logger
+from .libraries.vending_machine import VendingMachine
 
 ALLOWED_IMG_EXTENSIONS = {'pdf', 'png', 'jpg', 'jpeg', 'gif', 'svg'}
 ALLOWED_EXPERIMENT_EXTENSIONS = {'py'}
 IMAGE_UPLOAD_FOLDER = '/static/img'
 EXPERIMENT_UPLOAD_FOLDER = '/static/experiment'
 LOG_FOLDER = '/static/log'
 
-@APP.route('/run-trial', methods=['POST'])
-def run_trial():
-    """Responds with 'Running {trial_name}' string
+@APP.route('/run-experiment', methods=['POST'])
+def run_experiment():
+    """Start execution of experiment python file specified by user
 
-    All requests sent to this route should have a trial_name in
-    the query string, otherwise a 400 error will be returned
+    **Example request**:
 
-    Returns:
-        HTTP response 200 with body {"message":"Running <trial_name>"} or
-        HTTP response 400 with body {"message":"No trial_name specified"}
+    .. sourcecode::
 
-    """
+      POST /run-experiment?name=example_experiment HTTP/1.1
+      Host: localhost:5000
+      Accept-Encoding: gzip, deflate, br
+      Content-Length:
+      Connection: keep-alive
 
-    response = ""
+    **Example response**:
+
+    .. sourcecode:: http
+
+      HTTP/1.0 200 OK
+      Content-Type: application/json; charset=utf-8
+      Content-Length: 88
+      Server: Werkzeug/0.16.1 Python/3.8.1
+      Date: Thu, 13 Feb 2020 15:35:32 GMT
+
+      {
+        "log_file": "2020-03-17 05:15:06.558356 example_experiment.csv",
+        "message": "Running example_experiment"
+      }
+
+    All requests sent to this route should have an experiment file
+    included as a query parameter, otherwise a 400 error will be returned
+
+    :status 200: experiment started
+    :status 400: malformed request
+    """
+    response_message = ""
     response_code = 400
-    if request.args.get('trial_name') is not None:
-        trial_name = request.args.get('trial_name')
-        log_filename = str(datetime.utcnow()) + ' ' + trial_name + '.csv'
+    response_body = {}
+    if request.args.get('name') is not None:
+        experiment_name = request.args.get('name')
```

Comment:
See the way we passed filenames in the delete log endpoint, for homogeneity's sake you may want to consider doing it that way, or switching cases like those to use this method.

Reply:
Fixed in latest revision.